System/Console/ANSI/Color.hs

```diff
@@ -2,50 +2,59 @@
 module System.Console.ANSI.Color
   (
   -- | 8-bit to 24-bit color conversion
-    color8To24
+    xterm256ToSRGB
   -- | Utility functions
   , clamp
-  , color8CodeToEither
   , closest4bitsANSIColor
+  , closestXterm256Color
+  , color8CodeToXterm256
+  , colorToCode
+  , xterm256ColorToCode
     ) where
 
 
-import qualified Data.Colour.SRGB.Linear as SRGBLin(rgb, Colour)
-import qualified Data.Colour.SRGB as SRGB (RGB (..), toSRGB, Colour)
+import Data.Colour.SRGB (Colour, RGB (..), sRGB, toSRGB)
 
 import Data.Word (Word8)
 import Data.Colour.Names (black, blue, cyan, green, grey, lime, magenta, maroon,
                           navy, olive, purple, red, silver, teal, white, yellow)
 
 import Data.List (minimumBy, (!!))
+import Data.Maybe (fromJust)
 
 import System.Console.ANSI.Types
 
--- | Returns a color, in the linear sRGB colorspace, whose components are the components
---  of an 8-bit ANSI color.
+-- | Returns a color, in the linear sRGB colorspace, whose components are the
+-- components of an 8-bit ANSI color.
 -- For each color component, we first clamp the value to its admissible range
 -- defined in https://en.wikipedia.org/wiki/ANSI_escape_code#Colors:
 --  - 0-5 for 8-bit rgb
 --  - 0-23 for 8-bit grayscale
 -- Then we apply the xterm mapping (the mapped value is in range [0..255]).
 -- Finally we interpret the xterm-mapped values as components of a color
--- in linear srgb colorspace (we divide by 255 to express the value as a floating point in range [0 1]).
-color8To24 :: Color8 -> SRGBLin.Colour Float
-color8To24 (RGB8Color r' g' b') =
-  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapRGB8bitComponent . (\v -> clamp v 0 5) :: Word8 -> Float
-      r = mapping r'
-      g = mapping g'
-      b = mapping b'
-  in  SRGBLin.rgb r g b
-color8To24 (Gray8Color g') =
-  let mapping = (\v -> v/255.0) . fromIntegral . xtermMapGray8bitComponent . (\v -> clamp v 0 23) :: Word8 -> Float
-      g = mapping g'
-  in  SRGBLin.rgb g g g
+-- in linear srgb colorspace (we divide by 255 to express the value as a
```

Comment:
Here also if we chose srgb we can remove the "linear" word

Reply:
Agreed.